packages/pages-shared/__tests__/asset-server/handler.test.ts

```diff
@@ -1069,6 +1069,168 @@ describe("asset-server handler", () => {
 			);
 		});
 	});
+
+	test("should emit header when Web Analytics Token is injected", async () => {
+		const findAssetEntryForPath = async (path: string) => {
+			if (path === "/index.html") {
+				return "asset-key-index.html";
+			}
+			return null;
+		};
+
+		const fetchAsset = async () =>
+			Promise.resolve(
+				Object.assign(
+					new Response(`
+						<!DOCTYPE html>
+						<html>
+							<body>
+								<h1>Hello World</h1>
+							</body>
+						</html>
+					`),
+					{ contentType: "text/html" }
+				)
+			);
+
+		const { response } = await getTestResponse({
+			request: "https://example.com/",
+			metadata: createMetadataObject({
+				deploymentId: "mock-deployment-id",
+				webAnalyticsToken: "test-analytics-token",
+			}) as Metadata,
+			findAssetEntryForPath,
+			fetchAsset,
+		});
+
+		expect(response.status).toBe(200);
+		expect(response.headers.get("x-cf-pages-analytics")).toBe("1");
+
+		const responseText = await response.text();
+		expect(responseText).toContain(
+			'data-cf-beacon=\'{"token": "test-analytics-token"}\''
+		);
+	});
+
+	test("should not emit header when Web Analytics Token is not configured", async () => {
+		const findAssetEntryForPath = async (path: string) => {
+			if (path === "/index.html") {
+				return "asset-key-index.html";
+			}
+			return null;
+		};
+
+		const fetchAsset = async () =>
+			Promise.resolve(
+				Object.assign(
+					new Response(`
+						<!DOCTYPE html>
+						<html>
+							<body>
+								<h1>Hello World</h1>
+							</body>
+						</html>
+					`),
+					{ contentType: "text/html" }
+				)
+			);
```

Comment:
These two functions are duplicated across all the tests.
Perhaps move them outside and reuse?